lib/main.js

```diff
@@ -33,11 +41,19 @@ module.exports = {
   activate() {
     require('atom-package-deps').install('linter-jshint');
 
-    this.scopes = ['source.js', 'source.js-semantic'];
+    this.scopes = [];
+
+
     this.subscriptions = new CompositeDisposable();
+
+    this.subscriptions.add(atom.config.observe('linter-jshint.scopes', (value) => {
+      this.scopes = value;
```

Comment:
You **must** modify the array in place, you can't re-assign it. `linter` will be fed a reference from the initial call to `provideLinter` and will only ever check that first reference. Re-assigning the array will cause future changes to be ignored.

You can see a simple way of handling this [here](https://github.com/AtomLinter/linter-eslint/blob/v8.1.7/src/main.js#L75-L80).